quic/s2n-quic/src/provider/event/console_perf.rs

```diff
@@ -0,0 +1,296 @@
+        let receive_rate = rate(receive_progress, duration);
+        // The maximum congestion window observed during the interval
+        let max_cwnd = self.max_cwnd.swap(0, Ordering::Relaxed);
+        let max_cwnd = bytes(max_cwnd);
+        // The maximum amount of unacknowledged data in flight during the interval
+        let max_bytes_in_flight = self.max_bytes_in_flight.swap(0, Ordering::Relaxed);
+        let max_bytes_in_flight = bytes(max_bytes_in_flight);
+        // The maximum round trip time observed during the interval
+        let max_rtt = self.max_rtt.swap(0, Ordering::Relaxed);
+        let max_rtt = Duration::from_nanos(max_rtt);
+        // The maximum smoothed (weighted average) round trip time observed during the interval
+        let max_smoothed_rtt = self.max_smoothed_rtt.swap(0, Ordering::Relaxed);
+        let max_smoothed_rtt = Duration::from_nanos(max_smoothed_rtt);
+        // The number of packets recorded lost during the interval
+        let lost_packets = self.lost_packets.swap(0, Ordering::Relaxed);
+        // The number of event loop wakeups recorded during the interval
+        let wakeups = self.event_loop_wakeup.swap(0, Ordering::Relaxed);
+        // The duration of the latest event loop wakeup
+        let duration = self.timeout.swap(0, Ordering::Relaxed);
+        let duration = Duration::from_nanos(duration);
+        // The number of packet time out events observed during the interval
+        let pto_count = self.pto_count.swap(0, Ordering::Relaxed);
+        // The maximum rate at which packets are paced out observed during the interval
+        let max_pacing_rate = self.max_pacing_rate.swap(0, Ordering::Relaxed);
+        let max_pacing_rate = rate(max_pacing_rate, Duration::from_secs(1));
+        // The maximum estimate of bandwidth observed during the interval. Only output for BBRv2
+        let max_delivery_rate = self.max_delivery_rate.swap(0, Ordering::Relaxed);
+        let max_delivery_rate = rate(max_delivery_rate, Duration::from_secs(1));
+        println!(
+            "{send_rate}\t\
+            {receive_rate}\t\
+            {max_cwnd}\t\
+            {max_bytes_in_flight}\t\
+            {lost_packets}\t\
+            {wakeups}\t\
+            {duration:?}\t\
+            {max_rtt:?}\t\
+            {max_smoothed_rtt:?}\t\
+            {pto_count}\t\
+            {max_pacing_rate}\t\
+            {max_delivery_rate}",
+        );
+    }
+}
+
+fn rate(bytes: u64, duration: Duration) -> String {
+    use humansize::{format_size, DECIMAL};
+
+    let bits = bytes * 8;
+    let value = format_size(bits, DECIMAL.space_after_value(false));
+    let value = value.trim_end_matches('B');
+
+    if duration == Duration::from_secs(1) {
+        format!("{value}bps")
+    } else {
+        format!("{value}/{duration:?}")
+    }
+}
+
+fn bytes(value: u64) -> String {
+    use humansize::{format_size, DECIMAL};
+    format_size(value, DECIMAL.space_after_value(false))
+}
```

Comment:
It might be nice to make the "human" formatting configurable so it's easier for a script to process the raw data, rather than having to parse the suffixes

Reply:
I made a `TSV_RAW` format that does not print the suffixes